### PR TITLE
DEV-3746 | Refactor stripe webhook creation to support customer billing webhooks

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

This is PR 2 of 5 stacked PRs to fully implement server-side support for self-upgrade webhook events.

1. #1174 
2. #1178  (this PR)
3. #1177
4. #1185
5. #1176 

**NOTE** In short term, this PR  has `DEV-3741` as its base, but that branch is itself going to be merged into DEV-1750, which is the intended  base for this PR.  Once #1174 has been merged, DEV-1750 should be merged in here, and base should be DEv-1750.

#### What's this PR do?

This PR updates our code that handles Stripe webhook creation such that two separate endpoints are registered with Stripe, on two separate accounts:

| Endpoint  | Stripe account | Purpose |
| ------------- | ------------- | ---------- |
|  `/api/v1/revenginge-stripe-webhook/` |  NRH (NRE, Connect, legacy donations and client ACH) | Webhooks related to contributions |
|  `/api/v1/organizations/handle_stripe_webook/`| NRH (customer billing) | Making charges for self-upgrading orgs | 

The diff in this PR doesn't require much comment, save for one note: the `create_stripe_webhooks` command is moved from `apps/contributions/management` to `apps/common/management` to better align with its dual use spanning both organizations and contributions apps.

#### How should this be manually tested? Please include detailed step-by-step instructions.

This has to be tested by a dev.

1. Go into Stripe dashboard. Make sure you're in Test mode.
2. Select "NRH (customer billing)" account from dropdown on top left. Click "Developers => Webhooks" and ensure that you don't see a webhook called `https://dev-3746.revengine-review.org/api/v1/organizations/handle_stripe_webhook/`. If you do, delete it.
3. Select "NRH (NRE, Connect, legacy donations and client ACH)" account from dropdown on top left. Click "Developers => Webhooks" and ensure you don't see a webhook called `https://dev-3746.revengine-review.org/api/v1/revengine-stripe-webhook/`. If you do, delete it.
4. Go into the review app through `heroku run bash` then run `python manage.py create_stripe_webhooks`. You should see the webhook secret for two distinct accounts printed.
6. Back in the Stripe dashboard, go to the customer billing account and confirm you see an endpoint added for https://dev-3746.revengine-review.org/api/v1/organizations/handle_stripe_webhook/.
7. In the contributions account, confirm you see an endpoint added for  https://dev-3746.revengine-review.org/api/v1/revengine-stripe-webhook/

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No

#### Have automated unit tests been added? If not, why?

Yes

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

No

#### What are the relevant tickets? Add a link to any relevant ones.

- [DEV-3746](https://news-revenue-hub.atlassian.net/browse/DEV-3746)
- [DEV-1750](https://news-revenue-hub.atlassian.net/browse/DEV-1750)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

Yes.

- Add `STRIPE_TEST_SECRET_KEY_UPGRADES` env var (for both prod and in settings that get copied over in review apps once 1750 merged)
- After running the `create_stripe_webhooks` command in prod, you should add a value for `STRIPE_WEBHOOK_SECRET_FOR_UPGRADES`.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

1. If you created webhooks as part of testing this, please delete them in Stripe before continuing.
2. Copy over the env var changes requirements from this ticket to parent 1750 ticket.

[DEV-3746]: https://news-revenue-hub.atlassian.net/browse/DEV-3746?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by njharman — APPROVED** [review]
Looks good, AC met AFAICT

PR text just confirming wasn't typo with two different CONSTANTS

> Add STRIPE_TEST_SECRET_KEY_UPGRADES env var (for both prod and in settings that get copied over in review apps once 1750 merged)
> After running the create_stripe_webhooks command in prod, you should add a value for STRIPE_WEBHOOK_SECRET_FOR_UPGRADES.

**Review by x110dc — APPROVED** [review]
LGTM

**Review by x110dc — CHANGES_REQUESTED** [review]
There's an unresolved question for you (from me) on this one.

File: revengine/settings/base.py [hunk 1]
```diff
@@ -444,6 +445,10 @@ class CurrencyDict(TypedDict):
     "payment_method.attached",
     "invoice.upcoming",
 ]
+STRIPE_WEBHOOK_SECRET_FOR_UPGRADES = os.getenv("STRIPE_WEBHOOK_SECRET_FOR_UPGRADES", "")
+STRIPE_WEBHOOK_EVENTS_FOR_UPGRADES = [
+    "checkout.session.completed",
```
[the remark this answers is not in the document]
Comment (benjaminEwhite) [reply revengine/settings/base.py:450]: The `checkout.session.completed` event has a `client-reference-id` value that we control by virtue of configuring the pricing table. In DEV-3748, we add a `uuid` field to the organization model. We then set the client reference id in the Stripe pricing table in the SPA to the value of the org's UUID.

This provides what is hopefully a relatively clean and dependable way to link up an org's state with Stripe webhook events. 

In comparison, pairing a `customer.subscription.created/updated` with an org seems less clean and possibly more error prone. We'd end up having to do something like search for org by role assignment user emails, which seems kind of roundabout when we can cut straight to the chase setting `client-reference-id` in the pricing table.

When we bring in app, I think that's the moment to decide how to derive org's from initial `customer.subscription.created/updated` before which time the target org does not have a stripe subscription ID saved to it. It's true that we're guaranteed that we'll have to change later, but we get to kick the can down the road on some business logic that is not totally straight-forward, in favor of a quick clean interface in the short term that shouldn't paint us in any corners.

[the remark this answers is not in the document]
Comment (benjaminEwhite) [reply revengine/settings/base.py:450]: @x110dc here's the relevant places in DEV-3748:

- https://github.com/newsrevenuehub/rev-engine/blob/95087240b433bcfa6c57f62281ac3925cad7c393/apps/organizations/models.py#L125-L128
- https://github.com/newsrevenuehub/rev-engine/blob/95087240b433bcfa6c57f62281ac3925cad7c393/apps/organizations/views.py#L123

[the remark this answers is not in the document]
Comment (x110dc) [reply revengine/settings/base.py:450]: That's a great explanation and makes a lot of sense! I'd been wondering how we'd match up a payer if say, they paid with a different email address than what we have recorded for them. I guess this is one of the difficulties of reviewing stacked PRs in isolation.

**Pull request merged**